### ISSUE TITLE
Create Your First Mobile App with the Mobile Development Kit (MDK)

### ISSUE THREAD

**Issue by prtkkr** [SAPDocuments/Tutorials#4704]
Mission URL: https://developers.sap.com/mission.mobile-dev-kit-get-started.html

Hello Everyone,

Is there any way to upload data to back-end in the mobile app automatically whenever the app is online(i.e without clicking the sync button) ?

**Comment by jitendrakansal** [comment]
@prtkkr yes, you can. 
You can use the Connectivity module from Nativescript that we have also exposed via ClientAPI.nativescript.connectivityModule:
https://help.sap.com/viewer/977416d43cd74bdc958289038749100e/Latest/en-US/50ad902f4850457e912ab91b62c5e711.html
Doc:
https://docs.nativescript.org/ns-framework-modules/connectivity
 use
connectivityModule.startMonitoring
and
connectivityModule.stopMonitoring
functions to do that
It allows you to monitor when the app is connected to certain connection, and then you can trigger the upload/download

**Comment by prtkkr** [comment]
Thanks a lot for the help.

**Comment by prtkkr** [comment]
@jitendrakansal  I am new to sap and not able to figure out how to use the above connectivity module in the offline app. Could you please help me through a demo? Sorry for the trouble.

**Comment by jitendrakansal** [comment]
@prtkkr i would suggest that, you have a look at  this tutorial first about how to use one of the nativescript modules as it gives you idea how to write to create rule and call NS module. 

https://developers.sap.com/tutorials/cp-mobile-dev-kit-deep-link.html

And then try with connectivity module.

**Comment by MichaelCzcz** [comment]
Thank you for your feedback. The issue seems to be resolved, so I am closing the issue. 

If you still have questions, feel free to reopen the issue.

**Comment by prtkkr** [comment]
Hello @jitendrakansal , I have tried deep linking demo and implemented the same in my demo app and it is working fine. But the connectivity part is not working. In the tutorial, deep linking has been implemented on the "onClick" event of the buttons and I am unable to figure where do I have to implement the connectivity rule. It would be useful if you provide another tutorial implementing the connectivity module. Thank you for your help.
